addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1027,6 +1030,61 @@ options.registry.anchor = options.Class.extend({
     },
 });
 
+options.registry.CookiesBar = options.registry.SnippetPopup.extend({
+    xmlDependencies: [
+        '/website/static/src/xml/website.cookies_bar.xml',
+    ],
+
+    /**
+     * @constructor
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this.textButton = _t('I agree');
+        this.textTitle = _t('Respecting your privacy is our priority.');
+        this.textPrimary = _t('We use cookies to provide you a better user experience.');
+        this.textSecondary = _t('We use them to store info about your habits on our website. It will helps us to provide you the very best experience and customize what you see.') + '<br />' + _t('By clicking on this banner or navigating our website, you give us permission to collect data.');
```

Comment:
"or navigating our website" is not a valid consent according to GDPR, since the user may not have see the banner at all. 

For the record, the conditions for a consent to be valid are strict and listed in recital 32 of GDPR: https://gdpr-info.eu/recitals/no-32/

Here we are also missing a link to the Cookie Policy or Privacy Policy with cookie info.

Reply:
Also why are we harcoding so much text in JS, instead of putting it in a template. It's an anti-pattern ;)

Reply:
> Also why are we harcoding so much text in JS, instead of putting it in a template. It's an anti-pattern ;) It would be easy to have the default values in the template.

Actually, this all started with the 'I agree' button which handles the cookies bar closing (and set cookies etc).
Since the template is fully editable, the user can easily remove the 'I agree' button (by mistake or not). Thus, the goal here is to make sure there is a way to restore that button. Otherwise, the feature would be broken forever as no button to hide the cookies bar.

At the same time, I also added 3 different templates covering the more common Cookie Bars style.
This was the opportunity to use that layout option to reset the button on template change.

Then, I extended the behavior to the Cookies Bar title, main text, second text, so the user can always get back to the template default layout, without losing what he typed (also true when changing from one layout to another). For instance, the user customize layout 1 with its own text, then switch to layout 2, we don't want to override what he just wrote but integrate it in the new layout.

If there is a better way to store default value directly inside the templates, without duplicating the code, I'd be able to use it but in this case all 3 templates use the same text by default, and the same text to restore a remove element during layout change option.

Reply:
> We just discussed this with FP right now, using the Odoo.com Cookie Policy looks like a good compromise?
> At least, the user would have a complete feature ready to work, with a bootstraped Cookie Policy link/page.
> He will be able to edit it if he wants to.
> 
> What do you think?

Why not, hopefully they will get the hint and edit it according to their own use case. I guess it's enough to make it look like they are compliant, even if they don't know what they are doing.

Reply:
Yes I think so too.
Obviously, as you say, they don't know what they are doing, but they think they do.
People are 100% sure cookies bar are mandatory. They don't mind it or overthink it. Every 'figure d'autorité' even has cookies bars (gov website, police, banks..).
So yeah the feature is just to fit customer needs. They do need it, at least they think they need it, just gotta check number of ticket about it.